### ISSUE TITLE
Tempstore cannot find files for deletion

### ISSUE THREAD

**Issue by kainolophobia** [CollectionFS/Meteor-CollectionFS#669]
I'm uploading images to S3 and everything works, however I'm using the advice from [here](https://github.com/CollectionFS/Meteor-CollectionFS/wiki/How-to:-Customize-the-file-key---folders-in-the-S3-Bucket) to remove the mongo file id and collection name from the upload path. 

Upon doing so, I've started seeing errors of the following type:
`SA FileSystem: Could not delete /path/to/file/cfs/files/_tempstore/fakeCollectionName-mongoid-0.chunk because the file was not found.`

I'm assuming this is happening because I'm manually changing the fileKey attribute.

Everything appears to work, the image appears in S3 and there's no remnants on the machine (as far as I can tell).

Any idea how to fix this?

**Comment by rhyslbw** [comment]
I don't think tempstore uses the fileKeyMaker, but even if it did it shouldn't throw this error. 
https://github.com/CollectionFS/Meteor-CollectionFS/blob/master/packages/tempstore/tempStore.js

Do you have more than one app instances running?

Paste your fileKeyMaker function here

**Comment by kainolophobia** [comment]
I resolved my issue by dynamically creating a collection for each server instance; this way, the worker is never sent to clean up a file it can't see. 

For future readers, this will only work if you're using a collection as a Tempstore, where you don't need to keep track of the items on a collection basis (because each server instance is pointing to a different collection). In my use case, the actual store is S3 and all uploaded items are referenced elsewhere.

I believe the larger core problem has been addressed in the following issues:
[Update the cfs-worker](https://github.com/CollectionFS/Meteor-CollectionFS/issues/214)
[Upload from different apps and shared db](https://github.com/CollectionFS/Meteor-CollectionFS/issues/635)
